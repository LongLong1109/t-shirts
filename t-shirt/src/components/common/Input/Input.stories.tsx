import type { Meta, StoryObj } from '@storybook/react'

import Input from '@/components/common/Input'
import HiddenIcon from '@/icons/hiddenIcon'

const meta = {
  title: 'Components/Input',
  component: Input,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof Input>

export default meta
type Story = StoryObj<typeof Input>

export const InputPrimary: Story = {
  args: {
    value: 'Your Email',
    type: 'email',
    placeholder: 'Email address',
    onChange: () => {},
    icon: <HiddenIcon />,
  },
}
