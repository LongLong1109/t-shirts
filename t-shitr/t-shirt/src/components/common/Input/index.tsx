import { ChangeEvent } from 'react'
import Paragraph from '../Text'

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  value: string
  type: string
  placeholder: string
  onChange: (e: ChangeEvent<HTMLInputElement>) => void
  icon?: JSX.Element
  helperText?: string
  className?: string
  styleInput?: string
}

const Input = ({ icon, className, helperText, styleInput, ...rest }: InputProps) => {
  return (
    <div className={className}>
      <div className='relative flex items-center'>
        <input
          className={`outline-none px-6 py-3 bg-white-100 ${icon ? 'pr-10' : ''} ${styleInput}`}
          {...rest}
        />
        {icon && <div className='absolute right-3 text-dark-90'>{icon}</div>}
      </div>
      {helperText && (
        <Paragraph text={helperText} style='text-primary-100 dark:text-primary-100 pt-2' />
      )}
    </div>
  )
}

export default Input
