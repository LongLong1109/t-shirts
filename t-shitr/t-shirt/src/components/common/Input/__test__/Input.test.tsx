import { render, fireEvent } from '@testing-library/react';

// components
import Input from '@/components/common/Input';
import HiddenIcon from '@/icons/hiddenIcon';

describe('Input Component', () => {
  const onChange = jest.fn();
  it('renders the input with the provided value', () => {
    const { getByDisplayValue } = render(
      <Input value='test value' type='text' placeholder='Search...' onChange={onChange} />,
    );
    expect(getByDisplayValue('test value')).toBeInTheDocument();
  });

  it('renders the input with the icon when provided', () => {
    const { container } = render(
      <Input
        value=''
        type='text'
        placeholder='Search...'
        onChange={onChange}
        icon={<HiddenIcon />}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('calls the onChange handler when the input value changes', () => {
    const handleChange = jest.fn();
    const { getByPlaceholderText } = render(
      <Input value='' type='text' placeholder='Search...' onChange={handleChange} />,
    );
    const input = getByPlaceholderText('Search...');
    fireEvent.change(input, { target: { value: 'new value' } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });
});
