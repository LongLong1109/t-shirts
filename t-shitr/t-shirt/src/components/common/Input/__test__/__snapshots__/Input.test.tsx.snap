// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders the input with the icon when provided 1`] = `
<div>
  <div>
    <div
      class="relative flex items-center"
    >
      <input
        class="outline-none px-6 py-3 bg-white-100 pr-10 undefined"
        placeholder="Search..."
        type="text"
        value=""
      />
      <div
        class="absolute right-3 text-dark-90"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.119 15.6C14.0023 15.6 15.529 14.0733 15.529 12.19C15.529 10.3067 14.0023 8.78003 12.119 8.78003C10.2357 8.78003 8.70898 10.3067 8.70898 12.19C8.70898 14.0733 10.2357 15.6 12.119 15.6Z"
            stroke="#969696"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="10"
            stroke-width="1.2"
          />
          <path
            d="M12.1196 17.78C17.3774 17.78 21.6396 15.2772 21.6396 12.19C21.6396 9.1027 17.3774 6.59998 12.1196 6.59998C6.86186 6.59998 2.59961 9.1027 2.59961 12.19C2.59961 15.2772 6.86186 17.78 12.1196 17.78Z"
            stroke="#969696"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="10"
            stroke-width="1.2"
          />
          <path
            d="M3 21C11.7868 12.2132 12.2132 11.7868 21 3"
            stroke="#969696"
            stroke-linecap="round"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
